superagentx_handlers/websitecrawler/amazon.py

```diff
@@ -0,0 +1,190 @@
+                      'image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
+            'rtt': '200'
+        }
+        r = await self.asession.get(
+            url,
+            headers=headers
+        )
+        return BeautifulSoup(
+            r.html.html,
+            'html.parser'
+        )
+
+    def _get_deals(
+            self,
+            soup
+    ):
+        products = soup.find_all(
+            'div',
+            {
+                'data-component-type': 's-search-result'
+            }
+        )
+        if products:
+            for item in products:
+                title = item.find(
+                    'a',
+                    {
+                        'class': 'a-link-normal s-underline-text s-underline-link-text s-link-style a-text-normal'
+                    }
+                ).text.strip()
+                link = item.find(
+                    'a',
+                    {
+                        'class': 'a-link-normal s-underline-text s-underline-link-text s-link-style a-text-normal'
+                    }
+                ).get('href')
+                try:
+                    sale_prices = item.find_all('span', {'class': 'a-offscreen'})
+                    if sale_prices:
+                        sale_price = sale_prices[0].text.replace('$', '').replace(',', '').strip()
+                    else:
+                        sale_price = ''
+
+                    old_prices = item.find_all('span', {'class': 'a-offscreen'})
+                    if old_prices:
+                        old_price = old_prices[1].text.replace('$', '').replace(',', '').strip()
+                    else:
+                        old_price = ''
+                except:
+                    old_prices = item.find(
+                        'span',
+                        {
+                            'class': 'a-offscreen'
+                        }
+                    )
+                    old_price = float(
+                        old_prices.text.replace('£', '').replace(',', '').strip())
+                try:
+                    reviews = float(
+                        item.find(
+                            'span',
+                            {
+                                'class': 'a-size-base'
+                            }
+                        ).text.strip()
+                    )
+                except ValueError:
+                    logger.error("float error!...")
+
+                saleitem = {
+                    'title': title,
+                    'link': pyshorteners.Shortener(timeout=5).tinyurl.short('https://www.amazon.com' + link),
+                    'saleprice': sale_price,
+                    'oldprice': old_price,
+                    'reviews': reviews
+                }
+                self.deals_list.append(saleitem)
+
+    @staticmethod
+    async def _get_next_page(
+            soup
```

Comment:
Mention `type` annotation for this param

Reply:
fixed